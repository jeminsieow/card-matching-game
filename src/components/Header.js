import React from "react";
import { TouchableOpacity } from "react-native";
import styled from "styled-components/native";
import {
  COLOR_BLUE,
  COLOR_WHITE,
  HEADER_HEIGHT,
  HEADER_WIDTH,
} from "../utilities/constants";

const HeaderContainer = styled.View`
  height: ${HEADER_HEIGHT}px;
  width: ${HEADER_WIDTH}px;
  padding-bottom: 5px;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-end;
`;

const RestartButton = styled.Text`
  padding-left: 20px;
  padding-bottom: 5px;
  font-size: 20px;
  color: ${COLOR_BLUE};
`;

const StepsText = styled.Text`
  padding-right: 20px;
  font-size: 30px;
  color: ${COLOR_WHITE};
`;

const StepsNumber = styled.Text`
  font-size: 35px;
  color: ${COLOR_BLUE};
`;

export default function Header({ onPressResetGame, steps }) {
  return (
    <HeaderContainer>
      <TouchableOpacity onPress={onPressResetGame}>
        <RestartButton>Restart</RestartButton>
      </TouchableOpacity>
      <StepsText>
        STEPS: <StepsNumber>{steps}</StepsNumber>
      </StepsText>
    </HeaderContainer>
  );
}
