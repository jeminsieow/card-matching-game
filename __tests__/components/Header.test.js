import React from "react";
import { create } from "react-test-renderer";
import Header from "../../src/components/Header";

test("Header renders correctly", () => {
  const tree = create(
    <Header onPressResetGame={() => null} steps={5} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
