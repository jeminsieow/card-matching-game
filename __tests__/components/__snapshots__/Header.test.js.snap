// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 166.75,
        "justifyContent": "space-between",
        "paddingBottom": 5,
        "width": 710,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1B98F2",
            "fontSize": 20,
            "paddingBottom": 5,
            "paddingLeft": 20,
          },
        ]
      }
    >
      Restart
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 30,
          "paddingRight": 20,
        },
      ]
    }
  >
    STEPS: 
    <Text
      style={
        Array [
          Object {
            "color": "#1B98F2",
            "fontSize": 35,
          },
        ]
      }
    >
      5
    </Text>
  </Text>
</View>
`;
